near-sdk-sim/README.md

```diff
@@ -26,79 +32,229 @@ near-sdk = { git = "https://github.com/near/near-sdk-rs.git", tag="3.0.0" }
+    ...
 }
+
+#[near_bindgen]
+impl Token {
+    ...
+}
+```
+
+Then in your simulation test you can import `TokenContract`:
+
+```rust
+use token::TokenContract;
 ```
-Next we initialize the simulator and then use the deploy macro to deploy and 
-initialize a contract.  Next we use the call macro
+
+Now you can simplify the `init` & test code from the previous section:
+
 ```rust
+use near_sdk::serde_json::json;
+use near_sdk_sim::{DEFAULT_GAS, STORAGE_AMOUNT, call, deploy, init_simulator, to_yocto, view};
+
+fn init() {
+    let root = init_simulator(None);
+
+    let contract = deploy!(
+        contract: TokenContract,
+        contract_id: CONTRACT_ID,
+        bytes: &CONTRACT_WASM_BYTES,
+        signer_account: root,
+    );
+
+    (root, contract)
+}
+
+#[test]
+fn simulate_some_view_function() {
+    let (root, contract) = init();
+
+    let actual: String = view!(
+        contract.view_something("some_value".to_string()),
+    ).unwrap_json();
+
+    assert_eq!("expected", actual);
+}
+
 #[test]
-fn simple_test() {
-      let master_account = init_simulator(None);
-      let initial_balance = to_yocto("100000");
-          // uses default values for deposit and gas
-      let contract_user = deploy!(
-          // Contract Proxy
-          contract: FungibleTokenContract,
-          // Contract account id
-          contract_id: "contract",
-          // Bytes of contract
-          bytes: &TOKEN_WASM_BYTES,
-          // User deploying the contract,
-          signer_account: master_account,
-          // init method
-          init_method: new(master_account.account_id(), initial_balance.into())
-      );
-    let alice = master_account.create_user("alice".to_string(), to_yocto("100"));
-
-    let res = call!(
-        master_account,
-        contract.transfer(alice.account_id.clone(), transfer_amount.into()),
-        deposit = STORAGE_AMOUNT
+fn simulate_some_change_method() {
+    let (root, contract) = init();
+
+    let result = call!(
+        root,
+        contract.change_something("some_value".to_string()),
+        gas = DEFAULT_GAS / 2,
+        deposit = 0
     );
 
-    assert!(res.is_ok());
-    
-    let value = view!(contract.get_balance(master_account.account_id()));
-    let value: U128 = value.from_json_value().unwrap();
-    assert_eq!(initial_balance - transfer_amount, value.0);
+    assert!(result.is_ok());
 }
 ```
 
+# Common patterns
+
+## Profile gas costs
+
+TODO
+
+
+## Profile storage costs
+
+TODO
+
+
+## Inspect intermediate state of all calls in a complicated chain of transactions 
+
+TODO
+
+
+## Check expected transaction failures
+
+
+# Tweaking the genesis config
+
+For many simulation tests, using `init_simulator(None)` is good enough. This uses the [default genesis configuration settings](https://github.com/near/near-sdk-rs/blob/0a9a56f1590e1f19efc974160c88f32efcb91ef4/near-sdk-sim/src/runtime.rs#L59-L72).
+
+## Why would you want to change these?
+
+TODO
+
+## How do you change them?
+
+TODO
```

Comment:
@willemneal do you mind filling in this "Tweaking the genesis config" section? It doesn't need to be long.